querybook/server/app/auth/ldap_auth.py

```diff
@@ -1,76 +1,317 @@
-import re
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Dict, Iterator, List, Optional, Tuple, Union
 
 import ldap
 import flask_login
+import re
+from ldap.ldapobject import SimpleLDAPObject
 
 from flask_login import current_user
 from env import QuerybookSettings
 from app.db import with_session, DBSession
 from .utils import AuthenticationError, AuthUser, QuerybookLoginManager
+from lib.logger import get_logger
 from logic.user import (
     get_user_by_name,
     create_user,
 )
+from models.user import User
 
 login_manager = QuerybookLoginManager()
+logger = get_logger(__name__)
 
 
-def get_transformed_username(username):
-    dn = None
-    if username.startswith("uid="):
-        dn = username
+class AuthErrors:
+    BAD_CREDENTIALS = "Bad credentials"
+    UNAUTHORIZED = "Unauthorized"
 
-        match = re.match(r"^uid=([^,]+)", username)
-        username = match.group(1)
-    else:
-        dn = QuerybookSettings.LDAP_USER_DN.format(username)
-    return username, dn
 
+@dataclass
+class LDAPUserInfo:
+    LDAP_ATTRS = ["cn", "givenName", "mail", "sn"]
 
-@with_session
-def authenticate(username, password, session=None):
-    if not username or not password:
-        raise AuthenticationError()
+    cn: Optional[str] = None
+    dn: Optional[str] = None
+    email: Optional[str] = None
+    first_name: Optional[str] = None
+    last_name: Optional[str] = None
+
+    @property
+    def full_name(self) -> str:
+        full_name = " ".join(filter(bool, [self.first_name, self.last_name]))
+        return full_name if full_name else self.cn
+
+
+def _clear_ldap_search_results(
+    search_results: List[Tuple],
+) -> Optional[Tuple[str, Dict]]:
+    # Remove any search referrals from results
+    cleared_results = [
+        (dn, attrs)
+        for dn, attrs in search_results
+        if dn is not None and isinstance(attrs, dict)
+    ]
+    n_found = len(cleared_results)
+    if n_found != 1:
+        if n_found > 1:
+            logger.warning("LDAP search returned %s results" % n_found)
```

Comment:
use format string here?

Reply:
Ok 👍  I've aligned log messages to f-strings in this module. Also, I've changed `logger = get_logger(__name__)` to `LOG = get_logger(__file__)`, as it seems much more common in the repo.